solidity/random-beacon/contracts/libraries/Relay.sol

```diff
@@ -192,6 +192,27 @@ library Relay {
         return (inactiveMembers, slashingAmount);
     }
 
+    /// @notice Validates if the reporting of an unauthorized group signing can
+    ///         be performed. The signature verification is done by BLS lib.
+    function validateUnauthorizedSigning(
+        Data storage self,
+        bytes memory signedMsgSender,
+        Groups.Group memory group
+    ) internal {
+        require(isRequestInProgress(self), "No relay request in progress");
```

Comment:
Can you clarify why we need to tie unauthorized signing to the current relay entry? We don't do that in v1. If I understand correctly, we should allow to prove group private key leak at any moment, not just when that group is selected to handle a relay entry.

Reply:
I mixed up things, yea, I don't think it should be handled only for the current relay entry. c2d3c65
Tests to come..

Reply:
Fixing tests a part of: c1ebb30

Reply:
Ok good! Now if we don't deal with relay entry within `validateUnauthorizedSigning`, shouldn't we remove it from `Relay.sol` and do everything directly in `RandomBeacon`?

Reply:
Yea, I agree, I was on the fence with that, but I see you're also leaning towards moving it back, so here it is e8f169d